database/database.go

```diff
@@ -101,20 +103,31 @@ func Connect() error {
 		dbname=quote_gallery 
 		sslmode=disable`)
 	if err != nil {
-		return errors.New("Connect: connecting to database failed: " + err.Error())
+		return Status{
+			Code:    StatusError,
+			Message: "Connect: connecting to database failed: " + err.Error(),
+		}
 	}
 
-	return nil
+	return Status{
+		Code:    StatusOK,
+		Message: "",
+	}
 }
 
 // Initialize creates all the required tables in database, if they don't already exist
 // and initializes the cache from the database.
 //
 // Therefore it must be called before any other function of database.go despite Connect, which
-// needs to been have called for Initialize to work
-func Initialize() error {
+// needs to been have called for Initialize to work.
+//
+// Possible return states: StatusOK, StatusError
+func Initialize() Status {
 	if database == nil {
-		return errors.New("Initialize: not connected to database")
+		return Status{
+			Code:    StatusError,
```

Comment:
continuing my point from above - this is hardly more descriptive than returning a simple error.